### PR TITLE
small_change_and_add_error

### PR DESCRIPTION
**Pull request by Winnie08L**

### CODE REVIEW

File: surface_potential_analysis/kernel/plot.py [hunk 1]
```diff
@@ -500,3 +503,32 @@ def plot_isotropic_noise_kernel_1d_x(  # noqa: PLR0913
...
+    converted = convert_isotropic_kernel_to_basis(fitted_kernel, true_kernel["basis"])
+    return {
+        "basis": true_kernel["basis"],
+        "data": (converted["data"] - true_data) * 100 / true_data,
+    }
+
+
+def plot_isotropic_kernel_error(
+    true_kernel: IsotropicNoiseKernel[StackedBasisWithVolumeLike[Any, Any, Any]],
+    fitted_kernel: IsotropicNoiseKernel[StackedBasisWithVolumeLike[Any, Any, Any]],
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/plot.py:526]: support kernel of any basis, and use IsotropicNoiseKernel[_SBV0],

**Review by Matt-Ord — CHANGES_REQUESTED** [review]

**Comment by Matt-Ord** [comment]
Converted to draft for visibility, just click ready to review when it's ready for me

File: surface_potential_analysis/kernel/plot.py [hunk 2]
```diff
@@ -500,3 +504,32 @@ def plot_isotropic_noise_kernel_1d_x(  # noqa: PLR0913
...
+    true_data = true_kernel["data"].reshape(true_kernel["basis"].shape)
+    converted = convert_isotropic_kernel_to_basis(fitted_kernel, true_kernel["basis"])
+    return {
+        "basis": true_kernel["basis"],
+        "data": (converted["data"] - true_data) * 100 / true_data,
+    }
+
+
+def plot_isotropic_kernel_error(
+    true_kernel: IsotropicNoiseKernel[Any],
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/plot.py:526]: [_SBV0]

File: surface_potential_analysis/kernel/plot.py [hunk 3]
```diff
@@ -500,3 +504,32 @@ def plot_isotropic_noise_kernel_1d_x(  # noqa: PLR0913
...
+    fitted_kernel: IsotropicNoiseKernel[Any],
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/plot.py:527]: _SBV1

File: surface_potential_analysis/kernel/solve.py [hunk 4]
```diff
@@ -477,64 +477,6 @@ def get_noise_operators_real_isotropic_stacked_taylor_expansion(
     }
 
 
-def get_diagonal_noise_operators_eigenvalue(
```
Comment (Matt-Ord) [inline surface_potential_analysis/kernel/solve.py:480]: why was this deleted?

**Review by Matt-Ord — CHANGES_REQUESTED** [review]

**Comment by Matt-Ord** [comment]
We spoke about get_runtime in person - I'm not convinced this is the right approach, because it doesn't time the same function we actually use in practise

File: lib/sse_solver
[the remark this answers is not in the document]
Comment (Winnie08L) [reply lib/sse_solver:1]: but i didn't make any changes to sse_solver

[the remark this answers is not in the document]
Comment (Matt-Ord) [reply lib/sse_solver:1]: Ah so just make sure you revert this change then

File: surface_potential_analysis/kernel/solve.py (hunk 4 above)
Comment (Winnie08L) [reply surface_potential_analysis/kernel/solve.py:480]: below this there is a function called get_noise_operators_daigonal_eigenvalue and i think they are the same?and this deleted function is not called anywhere else

Comment (Matt-Ord) [reply surface_potential_analysis/kernel/solve.py:480]: Ah so there is - my bad

File: surface_potential_analysis/kernel/get_runtime.py
[the remark this answers is not in the document]
Comment (Winnie08L) [reply surface_potential_analysis/kernel/get_runtime.py:1]: i think this file is to time explicitly the runtime of each part of the functions to find the operation that takes the longest time, if we only want the total time of a complete fit method, then maybe don't need these functions, just need to time in system.py

**Review by Matt-Ord — APPROVED** [review]
